pkg/state/configmap_state.go

```diff
@@ -0,0 +1,125 @@
+package state
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"gopkg.in/yaml.v3"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/h8r-dev/heighliner/pkg/state/app"
+)
+
+// ConfigMapState state using k8s configmap as backend
+type ConfigMapState struct {
+	ClientSet *kubernetes.Clientset
+}
+
+// ListApps list all heighliner applications
+func (c *ConfigMapState) ListApps() ([]string, error) {
+
+	cms, err := c.ClientSet.CoreV1().ConfigMaps(HeighlinerNs).List(context.TODO(), metav1.ListOptions{
+		LabelSelector: labels.Set(map[string]string{configTypeKey: "heighliner"}).AsSelector().String(),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	apps := make([]string, 0)
+	for _, item := range cms.Items {
+		apps = append(apps, item.Name)
+	}
+	return apps, nil
+}
+
+// LoadOutput load output from configmap
+func (c *ConfigMapState) LoadOutput(appName string) (*app.Output, error) {
+
+	cm, err := c.ClientSet.CoreV1().ConfigMaps(HeighlinerNs).Get(context.TODO(), appName, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	if cm.Data["output.yaml"] == "" {
```

Comment:
In cmd/hln/cmd/variables.go, there is a cons `stackOutput  = "output.yaml"`.

Please export that one and use it here.

Reply:
Actually this is moved to `pkg/state/const.go` in this PR

Reply:
> Yes, as we can not use consts defined in cmd package from other package usually.